src/compare-page-pull-request/compare-page-pull-request.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+import { h } from 'dom-chef';
+import elementReady from 'element-ready';
+import { getRepoURL } from '../page-detect';
+import 'selector-observer';
+
+const Constants = {
+    COMPARE_DETAILS_SELECTOR: '#branch-compare.content-container'
+};
+
+export default async function init() {
+    await addUrlIfComparing();
+
+    /*
+     * Changes to the compare details selector indicate that:
+     *  1) The Compare button has been hit for the first time
+     *  2) The branches that are being compared have changed
+     */
+    document.body.observeSelector(
+        Constants.COMPARE_DETAILS_SELECTOR,
+        addUrlIfComparing
+    );
+}
+
+async function addUrlIfComparing() {
+    await elementReady(Constants.COMPARE_DETAILS_SELECTOR);
```

Comment:
Is this `await elementReady`  necessary? By the time this callback is executed by the `observeSelector` this node should already be in the DOM.